hack/hypernetes/util.sh

```diff
@@ -104,3 +108,27 @@ function kube::util::get_ip() {
 	ifconfig ${IF_NAME} | awk '/inet /{print $2}'
 }
 
+function kube::util::ensure_yum_ready() {
+	if pgrep yum 2>&1 1>/dev/null; then
+		rm -r /var/run/yum.pid
+	fi
+	sleep 3
+}
+
+function kube::util::clone_git_repo() {
+	local repo_url=$1
+	local repo_path=$2
+	if [[ -d  ${repo_path} ]] && (cd $repo_path && git status 2>&1 1>/dev/null) && [ ${DEV_MODE} = "y" ] ; then
+		echo "Git repo ${repo_path} already exist, leave it unchange in dev mode"
+	else
+		rm -rf ${repo_path}
+		git clone ${repo_url} ${repo_path}
+	fi
+}
+
+function kube::util::ensure_dir_exist() {
+	local dir=$1
+	if ! [[ -d ${dir} ]] ; then
```

Comment:
This check is actually unnecessary, `mkdir -p` will always returns success regardless of whether the dir is already exist.